src/ProductConstructionService/SubscriptionTriggerer/SubscriptionTriggerer.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.Serialization;
+using Azure.Storage.Queues;
+using Maestro.Data;
+using Maestro.Data.Models;
+using Microsoft.DotNet.DarcLib;
+using Microsoft.DotNet.Internal.Logging;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+
+namespace SubscriptionTriggerer;
+
+public class SubscriptionTriggerer
+{
+    private readonly IBasicBarClient _barClient;
+    private readonly OperationManager _operations;
+    private readonly ILogger<SubscriptionTriggerer> _logger;
+    private readonly BuildAssetRegistryContext _context;
+    private readonly QueueClient _queueClient;
+
+    public SubscriptionTriggerer(
+        ILogger<SubscriptionTriggerer> logger,
+        BuildAssetRegistryContext context,
+        IBasicBarClient barClient,
+        OperationManager operations,
+        QueueClient queueClient)
+    {
+        _operations = operations;
+        _logger = logger;
+        _context = context;
+        _barClient = barClient;
+        _queueClient = queueClient;
+    }
+
+    public async Task CheckSubscriptionsAsync(UpdateFrequency targetUpdateFrequency)
+    {
+        using (_operations.BeginOperation($"Updating {targetUpdateFrequency} subscriptions"))
+        {
+            var enabledSubscriptionsWithTargetFrequency = (await _context.Subscriptions
+                    .Where(s => s.Enabled)
+                    .ToListAsync())
+                    .Where(s => s.PolicyObject?.UpdateFrequency == targetUpdateFrequency);
```

Comment:
Why not put this `Where` inside of the query? We now query all subscriptions?

Reply:
I just copied this part, didn't make any changes, but I do agree the `where` after the `to list` looks a bit weird

Reply:
Maybe the value needs to be turned into an int or something though. Not sure how it's stored in a DB

Reply:
Oh damn, it's not a column 🤦🤦🤦